src/chromium/adding-third-party-crates/generating-gn-build-rules.md

```diff
@@ -0,0 +1,64 @@
+# Generating gn build rules
+
+A tool called `gnrt` converts from `third_party.toml` into the `BUILD.gn` rules
+required by our build system. It can also download crate source code.
+
+At the moment, this is a multi-step process - it may be streamlined in future.
+You'll need to mix:
+
+* generating build files
+* downloading missing crates
+
+and possibly run through the cycle a couple of times.
+
+To start, download the crate you want like this:
+
+```
+cd chromium/src
+vpython3 tools/crates/run_gnrt.py -- download CRATE_NAME CRATE_VERSION  --shipped yes --security-critical yes
```

Comment:
I assume Chromium engineers know what `vpython3` is?

Reply:
Um. I don't know what it _is_, but I have it installed as part of Chromium `depot_tools`, so I think it's fair to say all Chromies have it!

Reply:
Sounds good to me, as long as it's not a typo :smile: